gpytorch/lazy/mul_lazy_tensor.py

```diff
@@ -119,12 +120,17 @@ def _matmul(self, rhs):
 
         # Here we have a root decomposition
         if isinstance(self.left_lazy_tensor, RootLazyTensor):
-            left_root = self.left_lazy_tensor.root.evaluate()
+            if isinstance(self.left_lazy_tensor.root, DiagLazyTensor):
```

Comment:
I would want to see this tested out in practice. From the way that `MulLazyTensor` is recursively built, the size of the left/right vectors for `_quad_form_derivative` can grow exponentially unless you use a small root decomposition for all tensors. This is why we do a root decomposition on both the left and right tensors, rather than just one of them.

Reply:
I Reverted `MulLazyTensor` back to normal and special cased `__mul__` against `DiagLazyTensor` as well as `DiagLazyTensor.__mul__`.